### DISCUSSION TITLE
IgxOverlayService: set inputs/outputs of attached components

### DISCUSSION THREAD

**Discussion by Timmeeeey** [IgniteUI/igniteui-angular#10865] [Q&A] [answered]
## Is your feature request related to a problem? Please describe.
It is possible to attach a component via IgxOverlayService but you cannot set `@Input()` fields of the component.
Same with `@Output` fields.

## Describe the solution you'd like
It would be great if you could set the inputs/outputs of the component via an additional parameter in `IgxOverlayService.attach()`.

**Comment by wnvko** [comment, marked as answer]
Calling `attach` will fire `contentAppended` event. In the event arguments you have access to created `ComponentRef` and to the component you are trying to show in the overlay. Getting the component you can set any inputs or subscribe to any outputs you need like this:
```
    public ngOnInit(): void {
        this.overlayService.contentAppended.subscribe((e: OverlayEventArgs) => {
            const component = e.componentRef.instance as MyComponent;
            component.someInput = someValue;
            component.someOutput.subscribe(e => someSubscribtion);
        });
    }
```
[Here ](https://stackblitz.com/edit/angular-kczpnm?file=src%2Fapp%2Foverlay-main-1%2Foverlay-main-sample-1.component.ts)is a sample project showing how to set an input to the component.
